src/transformers/models/opt/modeling_flax_opt.py

```diff
@@ -0,0 +1,822 @@
+
+        outputs = [hidden_states, all_hidden_states, all_self_attns]
+
+        return outputs
+
+        if not return_dict:
+            return tuple(v for v in outputs if v is not None)
+
+        return FlaxBaseModelOutput(
+            last_hidden_state=hidden_states,
+            hidden_states=all_hidden_states,
+            attentions=all_self_attns,
+        )
+
+
+def make_positions(mask, padding_idx: int):
+    """Replace non-padding symbols with their position numbers.
+
+    Position numbers begin at padding_idx+1. Padding symbols are ignored.
+    """
+    positions = jnp.cumsum(mask, axis=1).astype(jnp.int32) + padding_idx
+    # positions = (jnp.cumsum(mask, axis=1) * mask).astype(jnp.int32) + padding_idx
+    return positions
+
+
+class FlaxOPTDecoder(nn.Module):
+    config: OPTConfig
+    dtype: jnp.dtype = jnp.float32  # the dtype of the computation
+    offset: int = 2
+
+    def setup(self):
+        self.dropout_layer = nn.Dropout(rate=self.config.dropout)
+
+        embed_dim = self.config.hidden_size
+        self.padding_idx = self.config.pad_token_id
+        self.max_target_positions = self.config.max_position_embeddings
+
+        # OPT is set up so that if padding_idx is specified then offset the embedding ids by 2
+        # and adjust num_embeddings appropriately. Other models don't have this hack
+        self.embed_tokens = nn.Embed(
+            self.config.vocab_size,
+            self.config.word_embed_proj_dim,
+            embedding_init=jax.nn.initializers.normal(self.config.init_std),
+        )
+        # TODO Check if that needs reimplemetation similar to OPTLearnedPositionalEmbedding
+        # should take attention mask as inputs ?
+        self.embed_positions = nn.Embed(
+            self.config.max_position_embeddings + self.offset,
+            embed_dim,
+            embedding_init=jax.nn.initializers.normal(self.config.init_std),
+        )
+
+        if self.config.word_embed_proj_dim != self.config.hidden_size:
+            self.project_in = nn.Dense(self.config.hidden_size, use_bias=False)
+            self.project_out = nn.Dense(self.config.word_embed_proj_dim, use_bias=False)
+
+        else:
+            self.project_in = None
+            self.project_out = None
+
+        self.layers = FlaxOPTDecoderLayerCollection(self.config, self.dtype)
+
+    def __call__(
+        self,
+        input_ids,
+        attention_mask,
+        position_ids,
+        head_mask=None,
+        init_cache: bool = False,
+        output_attentions: bool = False,
+        output_hidden_states: bool = False,
+        return_dict: bool = True,
+        deterministic: bool = True,
+    ):
+        input_shape = input_ids.shape
+        input_ids = input_ids.reshape(-1, input_shape[-1])
+
+        inputs_embeds = self.embed_tokens(input_ids)
+        if self.project_in is not None:
+            inputs_embeds = self.project_in(inputs_embeds)
+
+        positions = self.embed_positions(position_ids)
```

Comment:
I'd prefer to add a `+ self.offset` here instead of adding a +2 somewhat arbitrary when passing the position_ids

Reply:
Yes, I can change the +2 as a `+ self.offset` but should only be done when the positions ids are given no position ids are given (my understanding is that this is for testing). Since make position already computes the correct outputs )

Reply:
Yeah ideally, I'd actually move the +2 out of `make_position_ids` and just add it here (for both cases), like we do for BART. Not super important, but think this would be slightly more consistent and readable

Reply:
Updated it in PyTorch's OPT - think we can follow suite here